activerecord/lib/active_record/associations/belongs_to_association.rb

```diff
@@ -71,6 +75,8 @@ def replace(record)
           replace_keys(record)
 
           self.target = record
+
+          set_through_association_source_target(record)
```

Comment:
Should this be on the `target=` method rather than just the `replace` method?

Reply:
I'm not sure, do you think there would be any difference in practice? I was thinking that a through relationship will change only when replacing with another relationship, but I'm not sure. I was modelling it based on the fact that we create a join record when setting a `has_one :through` relationship in `replace` too, but I'm not sure if this should be different.